codegen/post_gen_hooks.go

```diff
@@ -23,114 +23,229 @@ package codegen
 import (
 	"fmt"
 	"path"
+	"path/filepath"
+	"sync"
 
+	"encoding/json"
 	"github.com/pkg/errors"
-	"path/filepath"
 )
 
+const clientInterface = "Client"
+
+// helper struct to pull out the fixture config
+type moduleConfig struct {
+	Config *config `json:"config"`
+}
+
+// config is the struct corresponding to the config field in client-config.json
+type config struct {
+	CustomImportPath string   `json:"customImportPath"`
+	Fixture          *Fixture `json:"fixture"`
+}
+
+// Fixture specifies client fixture import path and all scenarios
+type Fixture struct {
+	// ImportPath is the package where the user-defined Fixture global variable is contained.
+	// The Fixture object defines, for a given client, the standardized list of fixture scenarios for that client
+	ImportPath string `json:"importPath"`
+	// Scenarios is a map from zanzibar's exposed method name to a list of user-defined fixture scenarios for a client
+	Scenarios map[string][]string `json:"scenarios"`
+}
+
+// Validate the fixture configuration
+func (f *Fixture) Validate(methods map[string]interface{}) error {
+	if f.ImportPath == "" {
+		return errors.New("fixture importPath is empty")
+	}
+	for m := range f.Scenarios {
+		if _, ok := methods[m]; !ok {
+			return errors.Errorf("%q is not a valid method", m)
+		}
+	}
+	return nil
+}
+
 // ClientMockGenHook returns a PostGenHook to generate client mocks
-func ClientMockGenHook(h *PackageHelper) (PostGenHook, error) {
-	bin, err := NewMockgenBin(h.PackageRoot())
+func ClientMockGenHook(h *PackageHelper, t *Template) (PostGenHook, error) {
+	bin, err := NewMockgenBin(h, t)
 	if err != nil {
 		return nil, errors.Wrap(err, "error building mockgen binary")
 	}
 
 	return func(instances map[string][]*ModuleInstance) error {
 		fmt.Println("Generating client mocks:")
 		mockCount := len(instances["client"])
+		ec := make(chan error, mockCount)
+		var files sync.Map
```

Comment:
I assume you are bumping the Go version to 1.9 for access to this? If so can you adjust the pull request to note this for historical reasons?

Reply:
The bump is to get access to `sync.Map`, and I will put that in the PR description.